src/userSync.js

```diff
@@ -0,0 +1,169 @@
+import * as utils from 'src/utils';
+import { config } from 'src/config';
+
+export function newUserSync(userSyncConfig) {
+  let publicApi = {};
+  // A queue of user syncs for each adapter
+  // Let getDefaultQueue() set the defaults
+  let queue = getDefaultQueue();
+
+  // Since user syncs require cookie access we want to prevent sending syncs if cookies are not supported
+  let cookiesAreSupported = !utils.isSafariBrowser() && utils.cookiesAreEnabled();
+  // Whether or not user syncs have been trigger on this page load
+  let hasFired = false;
+  // How many bids for each adapter
+  let numAdapterBids = {};
+
+  // Use what is in config by default
+  userSyncConfig = userSyncConfig || config.getConfig('userSync');
+
+  /**
+   * @function getDefaultQueue
+   * @summary Returns the default empty queue
+   * @private
+   * @return {object} A queue with no syncs
+   */
+  function getDefaultQueue() {
+    return {
+      image: [],
+      iframe: []
+    };
+  }
+
+  /**
+   * @function fireSyncs
+   * @summary Trigger all user syncs in the queue
+   * @private
+   */
+  function fireSyncs() {
+    if (!userSyncConfig.syncEnabled || !cookiesAreSupported || hasFired) {
```

Comment:
If you're in a browser which doesn't support cookies, will the user syncs just build up inside the queue? Since this returns before calling `queue = getDefaultQueue()`. I'm not sure it's a big deal... but I suppose it could chew up memory if people run lots of auctions on the same page. Could probably wait for a secondary PR, though.

On that note... isn't this module basically a no-op if the browser doesn't support cookies? Again, perhaps a comment for a secondary PR, but... we could improve performance with something like:

```
export function newUserSync(userSyncDependencies) {
  if (!userSyncDependencies.cookiesAreSupported) {
    return {
      // Not pseudocode; these are literal no-ops
      registerSync: function() { }
      syncUsers: function() { }
      syncUsersOverride: function() { }
    }
  } else {
    // All the existing code & public API goes here
  }
}
```